### PR TITLE
Safe patch

### PR DESCRIPTION
**Pull request by detiber**
Use --api-version in patch commands and --output-version in any get commands where the output is compared or modified and api version would matter.

### CODE REVIEW

**Comment by detiber** [comment]
@sdodson I believe this covers all of our usage of patch and other oc commands that would be affected by version issues. ptal

@liggitt mind giving this a once over as well?

I did sneak in a minor additional change to clear the yum cache when modifying repos.

**Comment by liggitt** [comment]
the --api-version bits looks fine at first glance, did you run the individual commands to make sure output was as expected?

**Comment by detiber** [comment]
@liggitt I did the best as I could, I used a 3.0 deployment with --api-version=v1beta3 or --output-version=v1beta3 and all objects I tested with appeared to be correctly handled.

File: playbooks/common/openshift-cluster/upgrades/v3_0_to_v3_1/upgrade.yml [hunk 1]
```diff
@@ -517,24 +517,29 @@
     - _default_router.rc == 0
     - "'false' in _scc.stdout"
     command: >
-      {{ oc_cmd }} patch scc/privileged -p '{"allowHostPorts":true,"allowHostNetwork":true}' --loglevel=9
+      {{ oc_cmd }} patch scc/privileged -p
+      '{"allowHostPorts":true,"allowHostNetwork":true}' --loglevel=9
```
Comment (sdodson) [inline playbooks/common/openshift-cluster/upgrades/v3_0_to_v3_1/upgrade.yml]: Should we drop the loglevel option while we're here, probably an artifact from debugging when this was originally written?

**Comment by sdodson** [comment]
LGTM, looks like the repo cache commit should be a different PR though.

**Comment by detiber** [comment]
@sdodson rebased and removed the extraneous --loglevel

**Comment by sdodson** [comment]
Thanks, LGTM

**Comment by detiber** [comment]
rebased to resolve merge conflict.

@brenton

**Pull request merged**